File: go-service/models/sale.go
package models

import "time"

// Sale represents a sales transaction
type Sale struct {
	ID              int        `db:"Id" json:"id"`
	SaleNumber      string     `db:"SaleNumber" json:"saleNumber"`
	CustomerID      int        `db:"CustomerId" json:"customerId"`
	BranchID        int        `db:"BranchId" json:"branchId"`
	AttendantUserID int        `db:"AttendantUserId" json:"attendantUserId"`
	TotalAmount     float64    `db:"TotalAmount" json:"totalAmount"`
	PaymentMethod   string     `db:"PaymentMethod" json:"paymentMethod"`
	Status          string     `db:"Status" json:"status"`
	SaleDate        time.Time  `db:"SaleDate" json:"saleDate"`
	CreatedAt       time.Time  `db:"CreatedAt" json:"createdAt"`
	UpdatedAt       *time.Time `db:"UpdatedAt" json:"updatedAt,omitempty"`
}

// SaleItem represents an item in a sale
type SaleItem struct {
	ID         int       `db:"Id" json:"id"`
	SaleID     int       `db:"SaleId" json:"saleId"`
	ProductID  int       `db:"ProductId" json:"productId"`
	Quantity   int       `db:"Quantity" json:"quantity"`
	UnitPrice  float64   `db:"UnitPrice" json:"unitPrice"`
	TotalPrice float64   `db:"TotalPrice" json:"totalPrice"`
	Discount   *float64  `db:"Discount" json:"discount,omitempty"`
	CreatedAt  time.Time `db:"CreatedAt" json:"createdAt"`
}

File: frontend/src/app/core/domain/sale/sale.ts
import { Customer } from '../customer/customer';
import { Branch } from '../branch/branch';
import { User } from '../user/user';
import { PaymentMethod, OrderStatus } from '../../enums/enums.barrel';

/**
 * Represents a Sale Item
 */
export interface SaleItem {
  /** ID */
  id: number;

  /** Sale ID */
  saleId: number;

  /** Product ID */
  productId: number;

  /** Quantity */
  quantity: number;

  /** Unit Price */
  unitPrice: number;

  /** Subtotal */
  subtotal: number;

  /** Created At */
  createdAt?: Date;

  /** Updated At */
  updatedAt?: Date;
}

/**
 * Represents a Sale
 */
export interface Sale {
  /** ID */
  id: number;

  /** Sale Number - unique sale identifier */
  saleNumber: string;

  /** Customer ID - FK to Customer */
  customerId: number;

  /** Branch ID - FK to Branch */
  branchId: number;

  /** Attendant User ID - FK to User */
  attendantUserId: number;

  /** Total Amount */
  totalAmount: number;

  /** Payment Method */
  paymentMethod: PaymentMethod;

  /** Status */
  status: OrderStatus;

  /** Sale Date */
  saleDate: Date;

  /** Customer - associated customer */
  customer?: Customer;

  /** Branch - associated branch */
  branch?: Branch;

  /** Attendant User - user who processed the sale */
  attendantUser?: User;

  /** Items - sale items/line items */
  items?: SaleItem[];

  /** Created At */
  createdAt?: Date;

  /** Updated At */
  updatedAt?: Date;

  /** Deleted At */
  deletedAt?: Date | null;
}
